client.go

```diff
@@ -94,6 +101,26 @@ func NewClient(ctx context.Context, p OpenFGAParams) (*Client, error) {
 				ApiToken: p.Token,
 			},
 		}
+	} else {
+		config.Credentials = &credentials.Credentials{
+			Method: credentials.CredentialsMethodNone,
+		}
+	}
+	if p.HTTPClient != nil {
```

Comment:
Curiously, should we have the `else` check for ensuring that if a client is not provided, the default one is used?